src/main/java/org/jahia/modules/contentmanager/configuration/URLResolverFactory.java

```diff
@@ -102,6 +103,9 @@ private String convertPathinfo(String pathInfo) {
         try {
             servlet =  pathParts[CM_SERVLET_IDX];
             JahiaSite jahiaSite = ServicesRegistry.getInstance().getJahiaSitesService().getSiteByKey(pathParts[SITE_KEY_IDX]);
+            if (jahiaSite == null) {
+                jahiaSite = ServicesRegistry.getInstance().getJahiaSitesService().getSiteByKey(pathParts[SITE_KEY_IDX], JCRSessionFactory.getInstance().getCurrentUserSession(Constants.EDIT_WORKSPACE));
```

Comment:
Can't we just replace the first call (at line 105) by this one then?

Reply:
yes, actually users should have access to the site in edit anyway.